tests/unit/lms/services/course_test.py

```diff
@@ -28,7 +29,7 @@ def test_it_does_nothing_if_theres_already_a_matching_row(
         self, ai_getter, pyramid_request, svc
     ):
         existing_course = factories.Course(
-            consumer_key=pyramid_request.lti_user.oauth_consumer_key,
```

Comment:
Previously this just created a `Course` with randomly-generated `consumer_key` string but did not create a corresponding `ApplicationInstance` that has that consumer key. `existing_course.application_instance` would be `None`. If you tried to save the course to the DB it would crash. _Except_ that there was a separate `application_instance` fixture (at the bottom of the file, with `autouse=True`) with the same `consumer_key` so it would happen to work.

I think the new version is better: create the `Course` and the `ApplicationInstance` with the matching `consumer_key` in one place.